modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -0,0 +1,1200 @@
+
+        // fails here! check the problem
+        preProcessedInputs.emplace_back(preprocessed);
+        homogenizedInputDims.emplace_back(homogenizedInputDims_);
+
+        ++inputIter;
+    }
+}
+
+void LayerEinsumImpl::parseEquation(const String& equation)
+{
+    // copy copy of an eqution, will be changed
+    String eq = equation;
+
+    // remove white spaces in the copy
+    eq.erase(std::remove_if(eq.begin(), eq.end(), ::isspace), eq.end());
+
+    // check if '->' - the output subscript label is present in the equation;
+    std::size_t arrow_idx = eq.find("->");
+    if (arrow_idx != std::string::npos)
+    {
+        // split left and righ hand sides of the equation
+        lhs_eq = eq.substr(0, arrow_idx);
+        rhs_eq = eq.substr(arrow_idx + 2);
+        explicitEquation = true;
+    } else {
+        lhs_eq = eq;
+    }
+
+    // split lhs_eq by ',' - comma and put all created token - splits
+    // into lhs_eq_tokens vector
+    String token, comma = ",";
+    size_t idx = 0;
+    while(idx != String::npos){
+        idx = lhs_eq.find(comma);
+        token = lhs_eq.substr(0, idx);
+        lhs_eq.erase(0, idx + comma.length());
+        lhs_eq_tokens.emplace_back(token);
+    }
+}
+
+
+void LayerEinsumImpl::calculateOutputShape()
+{
+
+    if (outputSize!= 1)
+    {
+        CV_Error(Error::StsError,
+        cv::format("Einsum layer should only have one output, currenly [%d]", outputSize));
+    }
+
+    // Traverse through each of the subscript labels within the output subscript.
+    bool middleOfEllipsis = false;
+    // int64_t ellipsisCharCount = 0;
+
+    subscriptIndicesToOutputIndices.resize(numLetterIndices, -1);
+
+    std::array<int, numOfLetters> outputLetterToCount;
+    outputLetterToCount.fill(0);
+
+    int outputDimCounter = 0;
+    for (auto letter : rhs_eq)
+    {
+        if(letter == '.')
+        {
+            CV_Error(Error::StsNotImplemented, "Ellipsis are not supported yet");
+        } else {
+            if (middleOfEllipsis)
+            {
+                CV_Error(Error::StsError, "Encountered '.' character that is"
+                " not part of output subscript");
+            }
+
+            auto letterIndex = letterToIndex(letter);
+
+            if (letterIndex == -1)
+            {
+                 CV_Error(Error::StsError,
+                    "The only permissible subscript labels are"
+                    " lowercase letters (a-z) and uppercase letters (A-Z).");
+            }
+
+            if (outputLetterToCount[letterIndex] != 0)
+            {
+                CV_Error(Error::StsError,
+                 "Output subscript constains repeated letters");
+            }
+
+            ++outputLetterToCount[letterIndex];
+
+            auto mappedIndex = letter2index[letterIndex];
+
+            if(mappedIndex == -1)
+            {
+                CV_Error(Error::StsError,
+                "Output subscript has letters that were not encountered in the inputs");
+            }
```

Comment:
CV_Assert.

Reply:
Even better `CV_CheckNE(mappedIndex, -1, "Output subscript has letters that were not encountered in the inputs");`